order/views.py

```diff
@@ -3,25 +3,93 @@
 from django.views import  View
 from django.http import HttpResponse, JsonResponse
 from .models import Cart, Order, OrderStatus
-from product.models import Product
+from product.models import Product, ProductOption, Tag
 from account.utils import login_decorator
-
+    
 
 class CartView(View):
-   @login_decorator
-   def post(self,request):
+    @login_decorator
+    def post(self,request):
         data = json.loads(request.body)
         try:
-           cart_item =  Product.objects.select_related('category','collection','texture','stock_status').prefetch_related('tag','handle','wheel').get(product_number = data['product_number'])
-
-           Order.objects.create(
-               user_id = request.user.id,
-               order_status_id = 1
-           )
-           Cart.objects.create(
-               product_id  = cart_item.id,
-               amount      = data['amount']
-           )
-           return HttpResponse( status = 200 )
+            cart_item =  Product.objects.select_related('category','collection','texture','stock_status').prefetch_related('tag','cart_set','cart_set__order').get(product_number = data['product_number'])
+            if "tag" in data:
+                ProductOption.objects.create(
+                    product_id  = cart_item.id,
+                    tag_id      = Tag.objects.get(color = data['tag']).id,
+                    tag_text    = data['tag_text']
+                )
+            if Cart.objects.filter( product_id = data['product_id']).exists():
+                item = Cart.objects.get(product_id = data['product_id'])
+                item.amount += 1
+                item.save()
+                return HttpResponse(status = 200)
+            else:
+                order_item = Order.objects.create(
+                    user_id = request.user.id,
+                    order_status_id = 1
+                )
+                Cart.objects.create(
+                product_id      = cart_item.id,
+                amount          = data['amount'],
+                tag_id          = Tag.objects.get(color = data['tag']).id,
+                order_id     = order_item.id
```

Comment:
이미 만들어진 instance 들을 그대로 사용하셔도 됩니다.
```suggestion
                order   = order_item
```